src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -83,27 +161,266 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
+                                      TraceLoggingUInt64(peasantID, "peasantID", "Our peasant ID"),
+                                      TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+                    // Connect to the new monarch, which might be us!
+                    // If we become the monarch, then we'll return true and exit this thread.
+                    exitThreadRequested = _performElection();
+                    break;
+
+                case WAIT_OBJECT_0 + 1: // waits[1] was signaled, our manual interrupt
+
+                    TraceLoggingWrite(g_hRemotingProvider,
+                                      "WindowManager_MonarchWaitInterrupted",
+                                      TraceLoggingUInt64(peasantID, "peasantID", "Our peasant ID"),
+                                      TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+                    exitThreadRequested = true;
+                    break;
+
+                case WAIT_TIMEOUT:
+                    // This should be impossible.
+                    TraceLoggingWrite(g_hRemotingProvider,
+                                      "WindowManager_MonarchWaitTimeout",
+                                      TraceLoggingUInt64(peasantID, "peasantID", "Our peasant ID"),
+                                      TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+                    exitThreadRequested = true;
+                    break;
+
+                default:
+                {
+                    // Returning any other value is invalid. Just die.
+                    const auto gle = GetLastError();
+                    TraceLoggingWrite(g_hRemotingProvider,
+                                      "WindowManager_WaitFailed",
+                                      TraceLoggingUInt64(peasantID, "peasantID", "Our peasant ID"),
+                                      TraceLoggingUInt64(gle, "lastError", "The result of GetLastError"),
+                                      TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+                    ExitProcess(0);
+                }
+                }
+            }
+            catch (...)
+            {
+                // Theoretically, if window[1] dies when we're trying to get
+                // it's PID we'll get here. If we just try to do the election
+                // once here, it's possible we might elect window[2], but have
+                // it die before we add ourselves as a peasant. That
+                // _performElection call will throw, and we wouldn't catch it
+                // here, and we'd die.
+
+                // Instead, we're going to have a resilient election process.
+                // We're going to keep trying an election, until one _doesn't_
+                // throw an exception. That might mean burning through all the
+                // other dying monarchs until we find us as the monarch. But if
+                // this process is alive, then there's _someone_ in the line of
+                // succession.

```

Comment:
This seems to be a very common thing in this PR. We have a lot of loops that look like they could go on forever, but architecturally, they can't. I'm guessing a timeout wouldn't be good here?